src/http/stream.ts

```diff
@@ -448,30 +448,37 @@ async function decodePipelineResponse(
     if (encoding === "json") {
         const respJson = await resp.json();
         return readJsonObject(respJson, json_PipelineRespBody);
-    } else if (encoding === "protobuf") {
+    }
+
+    if (encoding === "protobuf") {
         const respData = await resp.arrayBuffer();
         return readProtobufMessage(new Uint8Array(respData), protobuf_PipelineRespBody);
-    } else {
-        throw impossible(encoding, "Impossible encoding");
     }
+
+    await resp.body?.cancel();
```

Comment:
do we need to await?

Reply:
I think we should. This will make sure exceptions are properly propagated and that things are correctly cleaned up before we move forward.